create-overlap-view fails

```
starting command: ant -v -Daction=create-overlap-view


BUILD FAILED
/SAN_humanmine/code/intermine/imbuild/postprocess.xml:31: error running PostProcessTask (action: create-overlap-view)
        at org.intermine.task.PostProcessTask.doCorePostProcess(PostProcessTask.java:148)
        at org.intermine.task.PostProcessTask.execute(PostProcessTask.java:121)
        at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:291)
        at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:601)
        at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
        at org.apache.tools.ant.Task.perform(Task.java:348)
        at org.apache.tools.ant.Target.execute(Target.java:390)
        at org.apache.tools.ant.Target.performTasks(Target.java:411)
        at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1399)
        at org.apache.tools.ant.Project.executeTarget(Project.java:1368)
        at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
        at org.apache.tools.ant.Project.executeTargets(Project.java:1251)
        at org.apache.tools.ant.Main.runBuild(Main.java:809)
        at org.apache.tools.ant.Main.startAnt(Main.java:217)
        at org.apache.tools.ant.launch.Launcher.run(Launcher.java:280)
        at org.apache.tools.ant.launch.Launcher.main(Launcher.java:109)
Caused by: java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:601)
        at org.intermine.task.PostProcessTask.doCorePostProcess(PostProcessTask.java:146)
        ... 17 more
Caused by: Operation was:create-overlap-view
        at org.intermine.bio.postprocess.PostProcessOperationsTask.execute(PostProcessOperationsTask.java:274)
        ... 22 more
Caused by: org.postgresql.util.PSQLException: ERROR: "overlappingfeaturessequencefeature" is not a view
  Hint: Use DROP TABLE to remove a table.
        at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2161)
        at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:1890)
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:255)
        at org.postgresql.jdbc2.AbstractJdbc2Statement.execute(AbstractJdbc2Statement.java:559)
        at org.postgresql.jdbc2.AbstractJdbc2Statement.executeWithFlags(AbstractJdbc2Statement.java:403)
        at org.postgresql.jdbc2.AbstractJdbc2Statement.executeUpdate(AbstractJdbc2Statement.java:331)
        at com.zaxxer.hikari.proxy.StatementJavassistProxy.executeUpdate(StatementJavassistProxy.java)
        at org.intermine.bio.postprocess.OverlapViewTask.createView(OverlapViewTask.java:84)
        at org.intermine.bio.postprocess.PostProcessOperationsTask.execute(PostProcessOperationsTask.java:257)
        ... 22 more

Total time: 12 seconds
Tue Dec  9 01:27:17 GMT 2014

finished


ERROR: 256
failed with exit code 0: ant -v -Daction=create-overlap-view
```

--- a/bio/postprocess/main/src/org/intermine/bio/postprocess/OverlapViewTask.java
+++ b/bio/postprocess/main/src/org/intermine/bio/postprocess/OverlapViewTask.java
@@ -49,11 +49,10 @@ public class OverlapViewTask
         Database db = this.osw.getDatabase();
         Connection con = db.getConnection();
 
-        con.setAutoCommit(false);
+        // autocommit as we may fail DROP TABLE and use same connection for DROP VIEW
+        con.setAutoCommit(true);
 
-        String dropSql = "DROP VIEW overlappingfeaturessequencefeature";
         String viewSql;
-
         if (osw.getSchema().useRangeTypes()) {
             viewSql =
                     "CREATE VIEW overlappingfeaturessequencefeature "
@@ -80,15 +79,26 @@ public class OverlapViewTask
                     + " and doesn't have bioseg installed. Aborting.");
         }
 
-        Statement statement = con.createStatement();
-        statement.executeUpdate(dropSql);
-        statement.close();
+        // initially this is a table, need to try dropping table first, if the postprocess has been
+        // run before then it will be a view. We need to try dropping table first then view.
+        String dropSql = "DROP TABLE overlappingfeaturessequencefeature";
+        try {
+            Statement statement = con.createStatement();
+            statement.executeUpdate(dropSql);
+            statement.close();
+        } catch (SQLException e) {
+            // if the postprocess has already been run will be a view
+            dropSql = "DROP VIEW overlappingfeaturessequencefeature";
+            Statement statement = con.createStatement();
+            statement.executeUpdate(dropSql);
+            statement.close();
+        }
 
-        statement = con.createStatement();
+
+        Statement statement = con.createStatement();
         statement.executeUpdate(viewSql);
         statement.close();
 
-        con.commit();
         con.close();
     }
 }

